instrumentation/ratpack-1.4/library/src/main/java/io/opentelemetry/instrumentation/ratpack/RatpackHttpAttributesExtractor.java

```diff
@@ -33,24 +38,8 @@ protected String url(Request request) {
     if (publicAddress == null) {
       return null;
     }
-    return publicAddress
-        .builder()
-        .path(request.getPath())
-        .params(request.getQueryParams())
-        .build()
-        .toString();
-  }
-
-  @Override
-  protected String target(Request request) {
-    // Uri is the path + query string, not a full URL
-    return request.getUri();
-  }
-
-  @Override
-  @Nullable
-  protected String host(Request request) {
-    return null;
+    URI uri = publicAddress.get();
```

Comment:
Private method to get the public address (the simplification I was hoping for with my TODO didn't materialize...